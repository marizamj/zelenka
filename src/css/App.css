.container {
  position: relative;
  padding-bottom: 50px;
}

.btnContainer {
  text-align: center;
  margin-top: -50px;
}

.btnImg {
  width: 200px;
  cursor: pointer;
  margin: 20px;
}

.selected {
}

.formIframe {
  width: 100%;
  border: none;
}

@media only screen and (max-width: 1023px) {
  .btnContainer {
    margin-top: -20px;
  }

  .btnImg {
    width: 140px;
    margin: 10px;
  }
}
