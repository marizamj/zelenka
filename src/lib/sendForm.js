export default function(data) {
  return fetch("https://zelenka-back.now.sh/form", {
    method: "POST",
    mode: "cors",
    body: JSON.stringify(data),
    headers: { "Content-Type": "application/json; charset=utf-8" }
  });
}
