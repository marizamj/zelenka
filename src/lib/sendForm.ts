import { FormData } from "../types";

export default function(data: FormData) {
  if (process.env.NODE_ENV === "development") {
    return Promise.resolve(new Response());
  }

  return fetch("https://zelenka-back.now.sh/form", {
    method: "POST",
    mode: "cors",
    body: JSON.stringify(data),
    headers: { "Content-Type": "application/json; charset=utf-8" }
  });
}
