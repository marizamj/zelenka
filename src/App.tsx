import React from "react";
import { UnmountClosed as Collapse } from "react-collapse";
import "./css/App.css";
import sendForm from "./lib/sendForm";
import videos from "./lib/videos";

import {
  Error,
  Form,
  Header,
  Video,
  SocialMedia,
  ThankYou,
  Partners
} from "./components";

import btnPartners from "./images/btn-partners.png";
import btnAll from "./images/btn-all.png";
import spinner from "./images/spinner.png";
import { FormType, FormData } from "./types";

const App = () => {
  const buttons = React.useRef<HTMLDivElement>(null);
  const [state, setState] = React.useState({
    openForm: "partners" as FormType,
    error: false,
    formIsSent: false,
    loading: false
  });

  const updateState = (newState: {
    openForm?: FormType;
    error?: boolean;
    formIsSent?: boolean;
    loading?: boolean;
  }) => setState({ ...state, ...newState });

  const toggleForm = (type: FormType) =>
    updateState({ formIsSent: false, error: false, openForm: type });

  const handleSubmit = (data: FormData) => {
    updateState({ formIsSent: false, loading: true, error: false });

    if (buttons.current && window) {
      window.scrollTo({ top: buttons.current.offsetTop });
    }

    sendForm(data)
      .then(res =>
        updateState({ formIsSent: res.ok, loading: false, error: !res.ok })
      )
      .catch(error => {
        console.error(error);
        updateState({ formIsSent: false, loading: false, error: true });
      });
  };

  const { openForm, formIsSent, loading, error } = state;

  return (
    <div className="container">
      <Header />
      <SocialMedia />

      <div ref={buttons} className="btnContainer">
        <img
          src={btnPartners}
          alt="Партнерам"
          className={`btnImg ${openForm === "partners" && "selected"}`}
          onClick={() => toggleForm("partners")}
          role="button"
          data-testid="button-partners"
        />
        <img
          src={btnAll}
          alt="Всем"
          className={`btnImg ${openForm === "all" && "selected"}`}
          onClick={() => toggleForm("all")}
          role="button"
          data-testid="button-all"
        />
      </div>

      {error && <Error />}
      {formIsSent && !error && <ThankYou />}

      {loading && (
        <div className="loading">
          <p>Отправляем...</p>
          <img className="spinner" alt="Пожалуйста, подождите" src={spinner} />
        </div>
      )}

      <Collapse isOpened={!formIsSent && !error && !loading}>
        <Form type={openForm} handleSubmit={handleSubmit} />
      </Collapse>
      <Video videos={videos} />
      <Partners />
    </div>
  );
};

export default App;
