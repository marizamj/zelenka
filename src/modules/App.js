import React, { Component } from "react";
import "../css/App.css";

import Header from "./Header";
import Video from "./Video";
import SocialMedia from "./SocialMedia";
import BottleBtn from "./BottleBtn";
import Form from "./Form";

import leftBottle from "../images/left-bottle.png";
import rightBottle from "../images/right-bottle.png";

class App extends Component {
  state = { openForm: null };

  toggleForm = type =>
    this.setState({ openForm: this.state.openForm === type ? null : type });

  render() {
    const { openForm } = this.state;

    return (
      <div className="container">
        <Header />
        <SocialMedia />
        <div className="bottleBtnContainer">
          <BottleBtn
            onClick={() => this.toggleForm("partners")}
            image={leftBottle}
            up={openForm !== "partners"}
            left
          />
          <BottleBtn
            onClick={() => this.toggleForm("friends")}
            image={rightBottle}
            up={openForm !== "friends"}
            right
          />
        </div>
        {openForm && <Form type={openForm} />}
        <Video />
      </div>
    );
  }
}

export default App;
