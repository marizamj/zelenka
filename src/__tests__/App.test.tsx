import React from "react";
import { act, render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

jest.mock("../lib/sendForm", () => jest.fn());
import sendForm from "../lib/sendForm";

import App from "../App";

const mockedSendForm = sendForm as jest.Mock;

const mapToAttribute = (attribute: string) => (element: HTMLElement) =>
  element.getAttribute(attribute);
const changeInput = (input: HTMLElement, value: string) =>
  fireEvent.change(input, { target: { value } });

afterEach(() => {
  jest.clearAllMocks();
});

describe("App", () => {
  test("Matches snapshot", () => {
    const { container } = render(<App />);

    expect(container).toMatchSnapshot();
  });

  test("Renders form for partners by default", () => {
    const { getByTestId } = render(<App />);

    expect(getByTestId("form-partners")).toBeInTheDocument();
  });

  test("Renders form for all if switched", () => {
    const { getByTestId } = render(<App />);

    fireEvent.click(getByTestId("button-all"));

    expect(getByTestId("form-all")).toBeInTheDocument();
  });
});

describe("App flows", () => {
  test("Happy flow", async () => {
    await act(async () => {
      mockedSendForm.mockImplementation(() => Promise.resolve({ ok: true }));

      const { getByTestId, getByText } = render(<App />);
      const submit = getByTestId("submit");
      window.scrollTo = jest.fn();

      changeInput(getByTestId("field-name"), "test-name");
      changeInput(getByTestId("field-email"), "test-email");
      changeInput(getByTestId("field-phone"), "1234567");
      changeInput(getByTestId("field-workplace"), "test-company");
      changeInput(getByTestId("field-message"), "test-message");

      fireEvent.click(submit);
      await setTimeout(jest.fn(), 0);

      expect(
        getByText("Спасибо! Мы ответим вам в ближайшее время. Соня и Юля")
      ).toBeInTheDocument();
      expect(mockedSendForm).toBeCalledTimes(1);
      expect(window.scrollTo).toBeCalledTimes(1);
    });
  });

  test("Error", async () => {
    await act(async () => {
      mockedSendForm.mockImplementation(() => Promise.resolve({ ok: false }));

      const { getByTestId } = render(<App />);
      const submit = getByTestId("submit");
      window.scrollTo = jest.fn();

      changeInput(getByTestId("field-name"), "test-name");
      changeInput(getByTestId("field-email"), "test-email");
      changeInput(getByTestId("field-phone"), "1234567");
      changeInput(getByTestId("field-workplace"), "test-company");
      changeInput(getByTestId("field-message"), "test-message");

      fireEvent.click(submit);
      await setTimeout(jest.fn(), 0);

      expect(getByTestId("error")).toBeInTheDocument();
      expect(mockedSendForm).toBeCalledTimes(1);
      expect(window.scrollTo).toBeCalledTimes(1);
    });
  });
});
