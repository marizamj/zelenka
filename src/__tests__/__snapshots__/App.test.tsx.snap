// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Matches snapshot 1`] = `
<div>
  <div
    class="css-1yjfgfj-App"
  >
    <header
      class="css-r270z7-Header"
    >
      <div
        class="css-1mq611n-Header"
      />
      <div
        class="css-1twly28-Header"
      >
        <img
          alt="ЗЕЛЕНКА"
          class="css-3dul9o-Header"
          src="logo.png"
        />
        <p
          class="css-rdlycs-Header"
        >
          Показываем хирургию и медицину так, чтобы не тошнило
        </p>
      </div>
    </header>
    <div
      class="css-p9n527-SocialMedia"
    >
      <a
        href="https://www.youtube.com/channel/UCIxnXifkJRTPOIe4AQM_fBQ"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="YouTube"
          class="css-qlvzh9-SocialMedia"
          src="youtube.png"
          style="margin-right: 5.5em;"
        />
      </a>
      <a
        href="https://www.facebook.com/3elenka.online/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Facebook"
          class="css-qlvzh9-SocialMedia"
          src="facebook.png"
        />
      </a>
    </div>
    <div
      class="css-k5lcjt-App"
    >
      <img
        alt="Партнерам"
        class="css-e9bp7j-App"
        data-testid="button-partners"
        role="button"
        src="btn-partners.png"
      />
      <img
        alt="Всем"
        class="css-1dk3h66-App"
        data-testid="button-all"
        role="button"
        src="btn-all.png"
      />
    </div>
    <div
      class="ReactCollapse--collapse"
      style="height: auto; overflow: initial;"
    >
      <div
        class="ReactCollapse--content"
      >
        <form
          class="css-13tf4h1-Form"
          data-testid="form-partners"
        >
          <div
            class="css-19lds2-Form"
          >
            Если вы врач, представитель клиники или любой другой компании, связанной с медициной, мы можем сделать очередной выпуск вместе. Заполните анкету, и мы очень скоро свяжемся с вами.
          </div>
          <p
            class="css-zjqj7m-Form"
          >
            * - обязательные поля
          </p>
          <label
            for="field-name"
          >
            <p
              class="css-1xzd690"
            >
              Как вас зовут?
              <span
                class="css-430mmm"
              >
                *
              </span>
            </p>
            <input
              data-testid="field-name"
              id="field-name"
              name="name"
              required=""
              type="text"
              value=""
            />
          </label>
          <label
            for="field-email"
          >
            <p
              class="css-1xzd690"
            >
              Контактный e-mail
              <span
                class="css-430mmm"
              >
                *
              </span>
            </p>
            <input
              data-testid="field-email"
              id="field-email"
              name="email"
              required=""
              type="email"
              value=""
            />
          </label>
          <label
            for="field-phone"
          >
            <p
              class="css-1xzd690"
            >
              Телефон
              <span
                class="css-430mmm"
              >
                *
              </span>
            </p>
            <input
              data-testid="field-phone"
              id="field-phone"
              name="phone"
              required=""
              type="tel"
              value=""
            />
          </label>
          <label
            for="field-workplace"
          >
            <p
              class="css-1xzd690"
            >
              Где вы работаете? Не забудьте адрес сайта.
              <span
                class="css-430mmm"
              >
                *
              </span>
            </p>
            <input
              data-testid="field-workplace"
              id="field-workplace"
              name="workplace"
              required=""
              type="text"
              value=""
            />
          </label>
          <label
            for="field-message"
          >
            <p
              class="css-1xzd690"
            >
              О чем вы хотели бы рассказать в Зелёнке?
              <span
                class="css-430mmm"
              >
                *
              </span>
            </p>
            <textarea
              data-testid="field-message"
              id="field-message"
              name="message"
              required=""
            />
          </label>
          <label
            for="field-money"
          >
            <p
              class="css-1xzd690"
            >
              Сколько денег вы готовы потратить на спонсорский выпуск?
            </p>
            <input
              data-testid="field-money"
              id="field-money"
              name="money"
              type="text"
              value=""
            />
          </label>
          <label
            for="field-startDate"
          >
            <p
              class="css-1xzd690"
            >
              Когда хотите начать?
            </p>
            <input
              data-testid="field-startDate"
              id="field-startDate"
              name="startDate"
              type="text"
              value=""
            />
          </label>
          <input
            data-testid="submit"
            type="submit"
            value="Отправить"
          />
        </form>
      </div>
    </div>
    <div
      class="css-1xdu56b-Video"
    >
      <div
        class="css-fn2lyz"
      >
        <h2>
          Первый выпуск
        </h2>
        <iframe
          allow="autoplay; encrypted-media"
          allowfullscreen=""
          class="css-g1rpge"
          frameborder="0"
          src="https://www.youtube.com/embed/F4iUlW4LpE8"
          title="Первый выпуск"
        />
      </div>
      <div
        class="css-fn2lyz"
      >
        <h2>
          Второй выпуск
        </h2>
        <iframe
          allow="autoplay; encrypted-media"
          allowfullscreen=""
          class="css-g1rpge"
          frameborder="0"
          src="https://www.youtube.com/embed/RFvHCQDeYGE"
          title="Второй выпуск"
        />
      </div>
    </div>
    <div
      class="css-1pietir-Partners"
    >
      <h2>
        Уже работают с нами
      </h2>
      <div
        class="css-1h34w8k"
      >
        <span
          class="css-cz0j02"
        >
          Бар «Петров и  Васечка»
        </span>
        <a
          href="http://piv.pub/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Бар «Петров и  Васечка»"
            class="css-g4hlvx"
            src="logo-pv.png"
          />
        </a>
      </div>
      <div
        class="css-1h34w8k"
      >
        <span
          class="css-cz0j02"
        >
          Клиника лазерной коррекции зрения «Новый взгляд»
        </span>
        <a
          href="http://www.nlv.ru/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Клиника лазерной коррекции зрения «Новый взгляд»"
            class="css-g4hlvx"
            src="logo-nlv.png"
          />
        </a>
      </div>
      <div
        class="css-1h34w8k"
      >
        <span
          class="css-cz0j02"
        >
          Материал «Афиша-Daily» о нас
        </span>
        <a
          href="https://daily.afisha.ru/news/20920-vyshel-pervyy-vypusk-bloga-zelenka-tam-rasskazyvayut-o-hirurgii-tak-chtoby-ne-toshnilo/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Материал «Афиша-Daily» о нас"
            class="css-g4hlvx"
            src="logo-afisha.png"
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;
